File: src/market_abm/api/schemas/shock.py
# Назначение файла: Pydantic DTO REST shock API (Slice 8.1 / Spec 011 §8.4).
from __future__ import annotations

from typing import Literal

from pydantic import BaseModel, Field


class SimulationShockRequest(BaseModel):
    shock_type: Literal[
        "demand_crash",
        "demand_boom",
        "platform_fee_hike",
        "platform_fee_cut",
        "marketplace_promotion",
        "supply_shock",
    ]
    intensity: float = Field(default=1.0, gt=0.0, le=3.0)
    duration_ticks: int | None = Field(default=None, ge=1, le=10_000)
    scenario: Literal["mild", "standard", "severe"] | None = "standard"
    shock_mode: Literal["stochastic_regime", "fixed_duration"] | None = None


class SimulationShockResponse(BaseModel):
    status: Literal["queued"] = "queued"
    shock_type: str
    queue_depth: int

File: frontend/src/types/shock.ts
/** Shock command DTO (Spec 009 §3.2 / Spec 011 §8.4). */

export type CrisisScenario = "mild" | "standard" | "severe";

export type SimulationShockRequest = {
  shock_type:
    | "demand_crash"
    | "demand_boom"
    | "platform_fee_hike"
    | "platform_fee_cut"
    | "marketplace_promotion"
    | "supply_shock";
  intensity: number;
  duration_ticks?: number | null;
  scenario?: CrisisScenario | null;
  shock_mode?: "stochastic_regime" | "fixed_duration" | null;
};

export type SimulationShockResponse = {
  status: "queued";
  shock_type: string;
  queue_depth: number;
};
